web-ui/components/Editor.vue

```diff
@@ -57,10 +70,49 @@ export default {
       },
     };
   },
+  mounted: function () {
+    let windowWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+    this.percentInPx = windowWidth / 100;
+    window.addEventListener("mousemove", (e) => this.resizeEditor(e));
+    window.addEventListener("click", this.stopDrag);
+    window.addEventListener('resize', () => this.updateWindowDimensions());
+  },
+  beforeDestroy() {
+    window.removeEventListener("mousemove", (e) => this.resizeEditor(e));
+    window.removeEventListener("click", this.stopDrag);
+    window.removeEventListener('resize', () => this.updateWindowDimensions());
+  },
   methods: {
     handleTab(tab) {
-      this.tab = tab;
+      this.tab = tab; 
     },
+    updateWindowDimensions(e) {
+      let windowWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+      this.percentInPx = windowWidth / 100;
+    },
```

Comment:
We don't need this at all, we can just do the calculation on the fly rather than storing it in the component.